### DISCUSSION TITLE
Override ResetPasswordNotification

### DISCUSSION THREAD

**Discussion by warsemann** [filamentphp/filament#11028] [Help]
Hi, it s possible to override the ResetPasswordNotification ? 
I want to change the layout of send mail, i try to override the sendPasswordResetNotification method in my User model (follow laravel doc) but this method is not use in filament ?

Thanks !

**Comment by finalbytes** [comment]
I've the same problem.
It should be possible for all texts to be adapted to the engineer's wishes.

**Reply by finalbytes** [reply]
What you can also try is as follows:

create a nl.json file under the resources/lang folder.

and then put the translation in here.

{
    "Reset Password Notification": "Wachtwoord opnieuw instellen",
    "You are receiving this email because we received a password reset request for your account.": "U ontvangt deze e-mail omdat we een verzoek voor het opnieuw instellen van het wachtwoord voor uw account hebben ontvangen.",
    "Reset Password": "Wachtwoord opnieuw instellen",
    "This password reset link will expire in :count minutes.": "Deze link voor het opnieuw instellen van het wachtwoord vervalt over :count minuten.",
    "If you did not request a password reset, no further action is required.": "Als u geen wachtwoordreset heeft aangevraagd, is er geen verdere actie vereist.",
    "Verify Email Address": "Bevestig e-mailadres",
    "Please click the button below to verify your email address.": "Klik op de onderstaande knop om uw e-mailadres te verifiëren.",
    "If you did not create an account, no further action is required.": "Als u geen account heeft aangemaakt, hoeft u geen verdere actie te ondernemen."
}

**Reply by hizmarck** [reply]
I added the Regards, the hello and the if you are having.... keys:

For Laravel 11, lang/es_MX.json for my case:



```
{
    "Hello!": "¡Hola!",
    "Reset Password Notification": "Notificación de Restablecimiento de Contraseña",
    "You are receiving this email because we received a password reset request for your account.": "Estás recibiendo este correo porque recibimos una solicitud de restablecimiento de contraseña para tu cuenta.",
    "Reset Password": "Restablecer Contraseña",
    "This password reset link will expire in :count minutes.": "Este enlace de restablecimiento de contraseña expirará en :count minutos.",
    "If you did not request a password reset, no further action is required.": "Si no solicitaste un restablecimiento de contraseña, no se requiere ninguna acción adicional.",
    "Verify Email Address": "Verificar Dirección de Correo Electrónico",
    "Please click the button below to verify your email address.": "Por favor, haz clic en el botón de abajo para verificar tu dirección de correo electrónico.",
    "If you did not create an account, no further action is required.": "Si no creaste una cuenta, no se requiere ninguna acción adicional.",
    "Regards": "Saludos",
    "If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\ninto your web browser:": "Si tienes problemas para hacer clic en el botón \":actionText\", copia y pega la siguiente URL\nen tu navegador web:"
}
```

**Reply by Jelmerkeij** [reply]
Just to note, the "Regards" key should have a comma at the end:

```json
{
 "Regards,": "Saludos",
}
```

**Comment by moskoweb** [comment]
Hello, same problem, change the Notification in the User model, but Filament seems to continue using the Laravel default, instead of the custom one. Has anyone found a solution, other than just the translation file?

**Reply by moskoweb** [reply]
I managed to change it by following what is said in this post, you need to update and adapt the code to use the custom notification:
`Filament\Pages\Auth\PasswordReset\RequestPasswordReset`

https://laraveldaily.com/post/filament-customize-auth-emails-reset-password-user-verification

**Reply by warsemann** [reply]
hi, i already tried this but it doesn't work for me

**Comment by warsemann** [comment]
Hi, with help by this plugin : https://filamentphp.com/plugins/visual-builder-email-templates
I override RequestPasswordReset::class from ->passwordReset() method : 

```
class RequestPasswordReset extends \Filament\Pages\Auth\PasswordReset\RequestPasswordReset
 {
     public function request(): void
     {
         try {
             $this->rateLimit(2);
         } catch (TooManyRequestsException $exception) {
             $this->getRateLimitedNotification($exception)?->send();
             return;
         }

         $data = $this->form->getState();

         $status = Password::broker(Filament::getAuthPasswordBroker())->sendResetLink(
             $data,
             function (CanResetPassword $user, string $token): void {
                 if (! method_exists($user, 'notify')) {
                     $userClass = $user::class;
                     throw new Exception("Model [{$userClass}] does not have a [notify()] method.");
                 }
                 $tokenUrl = Filament::getResetPasswordUrl($token, $user);

                 /**
                 * Use our custom notification is the only difference.
                 */
                $user->notify(new ResetPasswordNotification($tokenUrl));
             },
         );

         if ($status !== Password::RESET_LINK_SENT) {
             Notification::make()
                 ->title(__($status))
                 ->danger()
                 ->send();

             return;
         }

         Notification::make()
             ->title(__($status))
             ->success()
             ->send();

         $this->form->fill();
     }
 }
```

**Comment by duartec2** [comment]
A possible workaround is to replace the notification binding in the Laravel container. 

In my case I needed to use a custom blade template for the password reset notification an it works for me.

1. Create a custon notification
`php artisan make:notification CustomFilamentResetPasswordNotification
`
2. Then in **_CustomFilamentResetPasswordNotification.php_** you can extend the Filament Notification: 
```php
namespace App\Notifications;

use Filament\Notifications\Auth\ResetPassword as BaseNotification;

class CustomFilamentResetPasswordNotification extends BaseNotification
{
    public function toMail($notifiable)
    {
        return (new \Illuminate\Notifications\Messages\MailMessage)
            ->subject('Restablece tu contraseña')
            ->view('emails.reset-password', [
                'url' => $this->url,  
                'user' => $notifiable,
            ]);
    }
}

```

3. In **_AppServiceProvider.php_**, replace the binding
```php
use Filament\Notifications\Auth\ResetPassword as FilamentResetPassword;
use App\Notifications\CustomFilamentResetPasswordNotification;

public function register(): void
{
    $this->app->bind(FilamentResetPassword::class, function ($app, $params) {
        return new CustomFilamentResetPasswordNotification($params['token']);
    });
}
```
4. Blade template **_views/emails/reset-password.blade.php_**
```blade
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Recuperar contraseña</title>
</head>
<body>
    <h2>Hola {{ $user->name ?? 'usuario' }}</h2>
    <p>Haz clic en el siguiente enlace para restablecer tu contraseña:</p>
    <p><a href="{{ $url }}">{{ $url }}</a></p>
    <p>Si no fuiste tú, puedes ignorar este correo.</p>
</body>
</html>
```
If you just want to change the mail message you can extend the method **_buildMailMessage_**.
